clap_complete/src/env/shells.rs

```diff
@@ -398,8 +398,31 @@ compdef _clap_dynamic_completer_NAME BIN"#
             if i != 0 {
                 write!(buf, "{}", ifs.as_deref().unwrap_or("\n"))?;
             }
-            write!(buf, "{}", candidate.get_value().to_string_lossy())?;
+            write!(
+                buf,
+                "{}",
+                Self::escape_value(&candidate.get_value().to_string_lossy())
+            )?;
+            if let Some(help) = candidate.get_help() {
+                write!(
+                    buf,
+                    ":{}",
+                    Self::escape_help(help.to_string().lines().next().unwrap_or_default())
+                )?;
+            }
         }
         Ok(())
     }
 }
+
+impl Zsh {
+    /// Escape help string
+    fn escape_help(string: &str) -> String {
+        string.replace('\\', "\\\\")
+    }
```

Comment:
Normally, I'm against extraneous functions being on a struct like this but it helps keep it scoped and allows us to have an easier parallel between different shells and their implementation because they can each have a `<shell>::escape_value` and the completion function can refer to `Self::escape_value` and make it easier to read the code